test/client_and_receiver_integration_test.exs

```diff
@@ -396,36 +433,44 @@ defmodule SenderAndReceiverIntegrationTest do
       ]
     end
 
-    @tag port: 8160
+    @tag port: 8161
     @tag verify: :verify_none
     @tag client_cert: ""
     @tag keyfile: ""
     test "does not verify client cert if verify none option is provided on receiver", ctx do
       make_call_and_assert_success(ctx, ctx.ack)
     end
 
-    @tag port: 8161
+    @tag port: 8162
     @tag client_cert: ""
     @tag keyfile: ""
-    @tag reason:
-           {:certificate_required,
-            'TLS client: In state connection received SERVER ALERT: Fatal - Certificate required\n'}
+
+    if :erlang.system_info(:otp_release) == '22' do
```

Comment:
This has to be done as per the changes around the case when  socket is `nil`, our return is more consistent, but the charlist (string) varies from OTP ver  to OTP ver (consistent between OTP 23 and 24, but diff on 22).

Reply:
If you like, we can ignore message matching by making following changes to these tests:
```
  describe "client cert validation" do
    setup ctx do
      verify = ctx[:verify] || :verify_peer
      allowed_clients = ctx[:allowed_clients] || []

      tls_options = [
        cacertfile: "tls/root-ca/ca_certificate.pem",
        certfile: "tls/server/server_certificate.pem",
        keyfile: "tls/server/private_key.pem",
        verify: verify
      ]

      transport_opts = %{tls: tls_options}

      {:ok, %{pid: receiver_pid}} =
        MLLP.Receiver.start(
          port: ctx.port,
          dispatcher: MLLP.EchoDispatcher,
          transport_opts: transport_opts,
          allowed_clients: allowed_clients
        )

      client_cert = ctx[:client_cert] || "tls/client/client_certificate.pem"
      keyfile = ctx[:keyfile] || "tls/client/private_key.pem"

      sender_tls_options = [
        verify: :verify_peer,
        cacertfile: "tls/root-ca/ca_certificate.pem",
        certfile: client_cert,
        keyfile: keyfile
      ]

      tls_alert = ctx[:reason] || []

      expected_error_reasons = [:einval, :no_socket, :closed] ++ tls_alert

      on_exit(fn -> MLLP.Receiver.stop(ctx.port) end)

      [
        receiver_pid: receiver_pid,
        sender_tls_options: sender_tls_options,
        expected_error_reasons: expected_error_reasons
      ]
    end

    @tag port: 8161
    @tag verify: :verify_none
    @tag client_cert: ""
    @tag keyfile: ""
    test "does not verify client cert if verify none option is provided on receiver", ctx do
      make_call_and_assert_success(ctx, ctx.ack)
    end

    @tag port: 8162
    @tag client_cert: ""
    @tag keyfile: ""

    @tag reason: [:handshake_failure, :certificate_required]
    test "no peer cert", ctx do
      make_call_and_assert_failure(ctx, ctx.expected_error_reasons)
    end

    @tag port: 8163
    test "accepts a peer cert", ctx do
      make_call_and_assert_success(ctx, ctx.ack)
    end

    @tag port: 8164
    @tag allowed_clients: ["client-1"]
    test "accepts peer cert for allowed client", ctx do
      make_call_and_assert_success(ctx, ctx.ack)
    end

    @tag port: 8165
    @tag allowed_clients: ["client-x", "client-y"]
    test "reject peer cert for unexpected clients", ctx do
      expected_error_reasons = [:einval, :closed]

      log =
        capture_log(fn ->
          make_call_and_assert_failure(ctx, expected_error_reasons)
        end)

      assert log =~ ":fail_to_verify_client_cert"
    end

    @tag port: 8166
    @tag client_cert: "tls/expired_client/client_certificate.pem"
    @tag keyfile: "tls/expired_client/private_key.pem"

    @tag reason: [:certificate_expired]
    test "reject expired peer cert", ctx do
      make_call_and_assert_failure(ctx, ctx.expected_error_reasons)
    end

    @tag port: 8167
    @tag client_cert: "tls/server/server_certificate.pem"
    @tag keyfile: "tls/server/private_key.pem"

    @tag reason: [:handshake_failure]
    test "reject server cert as peer cert", ctx do
      make_call_and_assert_failure(ctx, ctx.expected_error_reasons)
    end

    @tag port: 8168
    @tag allowed_clients: ["client-x", "client-1"]
    test "accept peer cert from multiple allowed clients", ctx do
      make_call_and_assert_success(ctx, ctx.ack)
    end

    defp make_call_and_assert_success(ctx, expected_result) do
      assert expected_result == start_sender_and_send(ctx)
    end

    defp make_call_and_assert_failure(ctx, expected_error_reasons) do
      reason =
        start_sender_and_send(ctx)
        |> case do
          {:error, %Error{reason: {:tls_alert, {reason, _}}}} -> reason
          {:error, %Error{reason: reason}} -> reason
        end

      assert reason in expected_error_reasons
      # assert expected_error_reasons |> Enum.any?(&match?(^reason, &1))
    end

    defp start_sender_and_send(ctx) do
      {:ok, sender_pid} =
        MLLP.Client.start_link("localhost", ctx.port, tls: ctx.sender_tls_options)

      MLLP.Client.send(
        sender_pid,
        HL7.Examples.wikipedia_sample_hl7() |> HL7.Message.new()
      )
    end
  end
```